test/e2e/fpgaadmissionwebhook/fpgaadmissionwebhook.go

```diff
@@ -42,52 +40,34 @@ func init() {
 func describe() {
 	f := framework.NewDefaultFramework("webhook")
 
-	webhookDeployPath, err := utils.LocateRepoFile(deployScript)
-	if err != nil {
-		framework.Failf("unable to locate %q: %v", deployScript, err)
-	}
-
-	getEnv := func() []string {
-		return append(os.Environ(), "KUBECONFIG="+framework.TestContext.KubeConfig)
-	}
-
 	ginkgo.It("mutates created pods to reference resolved AFs", func() {
-		ginkgo.By("deploying webhook")
-		_, _, err := framework.RunCmdEnv(getEnv(), webhookDeployPath, "--kubectl", framework.TestContext.KubectlPath, "--namespace", f.Namespace.Name)
-		framework.ExpectNoError(err)
-
 		checkPodMutation(f, "fpga.intel.com/d5005-nlb3-preprogrammed",
 			"fpga.intel.com/af-bfa.f7d.v6xNhR7oVv6MlYZc4buqLfffQFy9es9yIvFEsLk6zRg")
 	})
 
 	ginkgo.It("mutates created pods to reference resolved Regions", func() {
-		ginkgo.By("deploying webhook")
-		_, _, err := framework.RunCmdEnv(getEnv(), webhookDeployPath, "--kubectl", framework.TestContext.KubectlPath, "--namespace", f.Namespace.Name)
-		framework.ExpectNoError(err)
-
 		checkPodMutation(f, "fpga.intel.com/arria10.dcp1.0-nlb0-orchestrated",
 			"fpga.intel.com/region-ce48969398f05f33946d560708be108a")
 	})
 
 	ginkgo.It("mutates created pods to reference resolved Regions in regiondevel mode", func() {
-		ginkgo.By("deploying webhook")
-		_, _, err := framework.RunCmdEnv(getEnv(), webhookDeployPath, "--kubectl", framework.TestContext.KubectlPath, "--namespace", f.Namespace.Name)
-		framework.ExpectNoError(err)
-
 		checkPodMutation(f, "fpga.intel.com/arria10.dcp1.0",
 			"fpga.intel.com/region-ce48969398f05f33946d560708be108a")
 	})
 }
 
 func checkPodMutation(f *framework.Framework, source, expectedMutation v1.ResourceName) {
-	ginkgo.By("waiting for webhook's availability")
-	if _, err := e2epod.WaitForPodsWithLabelRunningReady(f.ClientSet, f.Namespace.Name,
-		labels.Set{"app": "intel-fpga-webhook"}.AsSelector(), 1 /* one replica */, 10*time.Second); err != nil {
-		framework.DumpAllNamespaceInfo(f.ClientSet, f.Namespace.Name)
-		kubectl.LogFailedContainers(f.ClientSet, f.Namespace.Name, framework.Logf)
-		framework.Failf("unable to wait for all pods to be running and ready: %v", err)
+	kustomizationPath, err := utils.LocateRepoFile(kustomizationYaml)
+	if err != nil {
+		framework.Failf("unable to locate %q: %v", kustomizationYaml, err)
 	}
 
+	ginkgo.By("deploying webhook")
+	utils.DeployFpgaWebhook(f, kustomizationPath)
+
+	ginkgo.By("deploying mappings")
+	framework.RunKubectlOrDie(f.Namespace.Name, "apply", "-n", f.Namespace.Name, "-f", filepath.Dir(kustomizationPath)+"/../mappings-collection.yaml")
```

Comment:
slightly offtopic but we should add a test case for "Note that the mappings are scoped to the namespaces they were created in and they are applicable to pods created in the corresponding namespaces." too at some point.

Reply:
Agree. The script-based deployment kinda prevented that as the mappings were hardcoded to be installed at the same namespace as the webhook. Will create an issue.

Reply:
thanks!